sys/log/full/src/log_fcb.c

```diff
@@ -32,6 +32,41 @@
 
 static int log_fcb_rtr_erase(struct log *log);
 
+/**
+ * Helper function to find start point for walking, given an offset.
+ * for a non-zero offset, instead of walking from the beginning,
+ * walk from the last applicable area, check first entry and compare
+ * against the given offset. Repeat this until an offset less than the
+ * given offset is found, start walking from there.
+ */
+static int
+fcb_walk_back_find_start(struct fcb *fcb, struct log *log, struct log_offset *log_offset, struct fcb_entry *fcb_entry)
+{
+    struct flash_area *fap;
+    struct log_entry_hdr hdr;
+    struct fcb_entry iter_entry;
+    int rc;
+
+    fap = fcb->f_active.fe_area;
+
+    for (hdr.ue_index = log_offset->lo_index; hdr.ue_index >= log_offset->lo_index; fap--) {
```

Comment:
**fap** can't be just decremented, it is circular buffer, sometimes it wraps around so fap should be checked for this condition.